### ISSUE TITLE
Messages are not being sent

### ISSUE THREAD

**Issue by agusvazquez** [bitrise-steplib/steps-slack-message#52]
Hello guys, I am working with bitrise and I am using this awesome step to publish messages for Slack and I did use it in the past without troubles with webhooks.

Unfortunately it is not working and I don't know why. 

I have enabled debug mode and it prints a this JSON:

```
[33;1mBoth API Token and WebhookURL is provided. Using the webhook url[0m
Request to Slack: {"channel":"#dev_notifications","text":"New version available. https://play.google.com/apps/internaltest/4701449134258367556","attachments":[{"fallback":"I forgot the trending json\n\nSee merge request 2ourista/hipjaunt-mobile-react-native!11","color":"#3bc3a3","pretext":"*Build Succeeded!*","author_name":"Franco Cerrutti","title":"Merge branch 'json_correction' into 'development'","text":"I forgot the trending json\n\nSee merge request 2ourista/hipjaunt-mobile-react-native!11","fields":[{"short":true,"title":"App","value":"hipjaunt-mobile-react-native"},{"short":true,"title":"Branch","value":"development"},{"short":true,"title":"Workflow","value":"test"}],"footer":"Bitrise","footer_icon":"https://github.com/bitrise-io.png?size=16","ts":1545419874,"actions":[{"style":"default","text":"View App","type":"button","url":"https://app.bitrise.io/app/991509aa7853b590"},{"style":"default","text":"View Build","type":"button","url":"https://app.bitrise.io/build/c98a67e551f979f0"}]}],"icon_url":"https://github.com/bitrise-io.png","link_names":true,"username":"Bitrise"}

[32;1m
Slack message successfully sent! 🚀
[0m
|                            
```

But if I copy that Json into the POSTMAN with the same webhook URL it works like charm, so I guess that is from your side or there is something I am doing wrong.

![success](https://user-images.githubusercontent.com/1704434/50359945-3ad9b180-053d-11e9-83ff-f9ec25d655e1.PNG)

Thanks for your support.

Cheers!

**Comment by bitce** [comment]
Hi @agusvazquez! Sorry to hear about the issue. Could you please include a build URL for us to take a look at as well?

**Comment by agusvazquez** [comment]
Hey bitce! Thanks for your reply.

Here is the build URL.

https://app.bitrise.io/build/7b11793a3f164155

Let me know if you need anything else.

Happy new year!

**Comment by fehersanyi-bitrise** [comment]
Hy @agusvazquez!

this may be a dumb question, but are you sure that the credencials are all ok? 
I meant the webhook, and it is set up correctly in slack as well?

**Comment by agusvazquez** [comment]
Hi @fehersanyi-bitrise ! Yes hahaha it is everything ok.

If not it wouldnt print on the logs "Slack message successfully sent! 🚀", right?

**Comment by fehersanyi-bitrise** [comment]
@agusvazquez and the setup is good in the slack side as well? because if it was an error on bitrise it would not succeed.

**Comment by agusvazquez** [comment]
Yes, because when i call the same request using postman with the webhook url with the JSON body that appears on the debug screen it works like charm.

**Comment by fehersanyi-bitrise** [comment]
summoning @godrei or @viktorbenei

**Comment by leodeleon22** [comment]
@fehersanyi-bitrise I'm also having this issue
https://app.bitrise.io/build/6097bbf7b75e6db8

**Comment by fehersanyi-bitrise** [comment]
Hy @leodeleon22 

You did not provide a channel to post to, that is the issue I believe.

**Comment by priscila225** [comment]
Hi, i found the problem, i had the same problem and spent a lot of time to figure out. Seems we can't have both API and slack webhook on this bitrise step configuration, when i removed the API and just leave the webhook worked! Don't know if is the same problem you have but finally found it.

**Comment by mhr78** [comment]
Wow! Thank you @priscila225 you are my hero!

**Comment by bitce** [comment]
Good job @priscila225! Note that this is printed out by the step too :)

**Comment by priscila225** [comment]
Hi @bitce for me this printed when both are **empty** as a error message but when I add both values, i see this message: "Both API Token and WebhookURL is provided. Using the webhook url" but this is not occuring, when i add API token and webhook url the step don't work. Maybe this is something missing to use webhook url when provide both? also don't know the reason why the API token stop to work. Hope to help :) thank you!

**Comment by bitce** [comment]
Hi @priscila225, I just gave this a try and I can confirm, thanks for bringing this to our attention! I'll let the team know :)

**Comment by glauberfc** [comment]
> Hi, i found the problem, i had the same problem and spent a lot of time to figure out. Seems we can't have both API token and slack webhook on this bitrise step configuration, when i removed the API token and just leave the webhook it worked fine! Don't know if it's the same problem you have but finally found it.

This solution worked to me. Thanks, @priscila225!

**Comment by ofalvai** [comment]
@chedabob thanks for the research and pointing it out, this has indeed been fixed in #55, we just forgot to update this thread.

We can close this issue, I believe it's better to not fail the build if both inputs are provided. If the provided API token is invalid, the step would fail anyway, directing the users' attention to the logs where they can see the warning about this issue.
